### ISSUE TITLE
SQL script with ;WITH

### ISSUE THREAD

**Issue by perverez** [dbeaver/dbeaver#15543]
Hello,

running the code listed as batch (Alt + X), displaying the error 'Must declare the table variable "@TABLENAME"'. Under MS SQL Server Management Studio, the script is correct. I believe the semicolon in front of WITH is causing the problem. The code is of course only an example.

Can you help me with the error? Thank you. P.

```
DECLARE @TABLENAME TABLE (ID UNIQUEIDENTIFIER)

;WITH CTE AS (
  SELECT ROWNUMBER = ROW_NUMBER() OVER (PARTITION BY ID ORDER BY ID) FROM @TABLENAME
)
DELETE FROM CTE WHERE ROWNUMBER > 1

```
Microsoft SQL Server Standard (64-bit)
Windows Server 2016 Standard (10.0)
Version: 15.0.2080.9

DBeaver: Version 21.3.4.202202051448

**Comment by E1izabeth** [comment]
Hello, @perverez. Thank you for bug report.
As a workaround you can select whole statement and press Ctrl+Enter to run it.

**Comment by Impaler67** [comment]
Yeah. This is still an issue... Spent half an hour looking for mistakes in my query
